### PR TITLE
Add bootloader control to support adb reboot xxx

### PR DESCRIPTION
**Pull request by xiaopingx**
For now, we can use
       adb reboot bootloader
       adb reboot-bootloader
       adb reboot fastboot
    boot into fastboot mode

       adb reboot recovery
    boot into recovery mode

Bug ID: https://01.org/jira/browse/AIA-159
Test: Test on Joule board, them could work well

Signed-off-by: hu,xiaopingx <xiaopingx.hu@intel.com>
Signed-off-by: Chen,Xihua <xihua.chen@intel.com>

### CODE REVIEW

**Comment by xiaopingx** [comment]
adb reboot bootloader
adb reboot-bootloader
adb reboot fastboot
adb reboot recovery
could work well Joule

**Comment by kalyankondapally** [comment]
@Kishore409 Can you test on joule

**Comment by xiaopingx** [comment]
https://github.com/android-ia/device-intel-common/pull/2
https://github.com/android-ia/system-core/pull/3
https://github.com/android-ia/bootable-recovery/pull/2
https://github.com/android-ia/device-androidia-mixins/pull/31
https://github.com/android-ia/device-androidia/pull/39

File: recovery/bootloader/include/bootloader_control.h [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+// C Interface.
+bool write_reboot_bootloader(void);
+#endif  // ifdef __cplusplus
+
+
+
+
+#endif
+
+
```
Comment (shobhitka) [inline recovery/bootloader/include/bootloader_control.h]: Can we remove the spurious new lines above

File: recovery/bootloader/bootloader_control.cpp [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+  }
+  /************************************************/
+ 
+ 
+ // strlcpy(boot.command, "bootonce-bootloader", sizeof(boot.command));
+  return write_bootloader_message(boot, err);
+}
+extern "C" bool write_reboot_bootloader(void) {
+  std::string err;
+  return write_reboot_bootloader(&err,0);
```
Comment (shobhitka) [inline recovery/bootloader/bootloader_control.cpp]: 0 ? thats not defined as a target in the header file below ?

**Pull request closed without merging**